src/amici_interface_matlab.cpp

```diff
@@ -376,7 +383,7 @@ ExpData *expDataFromMatlabCall(const mxArray *prhs[], const UserData *udata,
     int ne_mz = 0, nz_mz = 0, ne_sigmaz = 0,
         nz_sigmaz = 0; /* integers with problem dimensionality */
     ExpData *edata = new ExpData(udata, model);
-    if (edata->my == nullptr) {
+    if (edata->my.empty()) {
         // if allocation fails we throw an exception
```

Comment:
If allocation fails, std::bad_alloc should have been thrown already and we will never get here. I think we can remove this block.